### ISSUE TITLE
Add Developer-Signed F-Droid Release APK to Github/Accrescent

### ISSUE THREAD

**Issue by shrimprugbysnowowl** [organicmaps/organicmaps#9909]
There appears to be three android versions of Organic Maps that are available for installation. They are differentiated in the name with "google", "web", and "fdroid".

The "google" build is available in the google play and Accrescent stores, the "fdroid" build is available only from F-Droid, and the "web" build is available here on Github.

As detailed in #8747 , proprietary google libraries are included in the "google" and "web" builds. The "fdroid" build does not contain these libraries.

To most privacy and FOSS advocates, excluding proprietary libraries is highly valued as long as potential sacrifices to functionality are acceptable.

I suggest publishing the "fdroid" build to both the Accrescent App Store (replacing the current "google" build) and to Github and adding the sha-256 hash of the developer app signing key, which appears to be "B9:C7:AE:79:A5:A9:02:70:DF:08:A1:32:E5:36:B9:C6:66:F5:BE:F1:F5:9B:30:4F:CE:CF:86:87:86:5E:4B:5B", to the README.

**Comment by pastk** [comment]
Technically we can publish F-Droid flavors of OM releases, but they will be signed by OM's own keys, not F-Droid ones. But this would create confusion and apks from OM and F-Droid won't be interchangeable (i.e. it won't be possible to replace/upgrade from different source).

Alternatively we could add another flavor, e.g. OM FOSS (perhaps with its own app id), but this seems like a lot work to maintain a very small difference between flavors.

I think an overall better solution would be to gradually phase out the proprietary Google lib. A FOSS replacement was added to the F-Droid flavor recently #9575, if everything is good then we can do the Web flavor next.

**Comment by pastk** [comment]
> and adding the sha-256 hash of the developer app signing key, which appears to be "B9:C7:AE:79:A5:A9:02:70:DF:08:A1:32:E5:36:B9:C6:66:F5:BE:F1:F5:9B:30:4F:CE:CF:86:87:86:5E:4B:5B", to the README.

Open a PR to add e.g. a "Verifiability" section?

**Comment by pastk** [comment]
> I think an overall better solution would be to gradually phase out the proprietary Google lib.

- https://github.com/organicmaps/organicmaps/issues/9916

**Comment by shrimprugbysnowowl** [comment]
Agreed on phasing out the google libs! Great to hear about the fdroid version. I'll try to keep an eye out for that update to the web version. Do you think you could estimate a very rough/general timeline? Just want to use it to check back, not to bug anyone.

I'm happy to create the suggested PR.

-------- Original Message --------

**Comment by pastk** [comment]
> Do you think you could estimate a very rough/general timeline?

It could happen faster if there is a volunteer to make it happen :)
Personally I'm unlikely to do it anytime soon.

**Comment by shrimprugbysnowowl** [comment]
I can certainly try, but wouldn't want my efforts to cause more frustration than help. Do you know the PR offhand that eliminated google libs from the f-droid version?

-------- Original Message --------

**Comment by pastk** [comment]
Its https://github.com/organicmaps/organicmaps/pull/9575
(there was no Google location lib in the F-Droid flavor, the PR added a microG implementation)

**Comment by shrimprugbysnowowl** [comment]
Perfect, thank you! I'll see if I can figure this out

-------- Original Message --------

**Comment by pastk** [comment]
Accrescent serves apks signed by app devs.

IDK if they have enabled automatic fetching of new releases already. 
AFAIK current uploads were done manually by @biodranik, right?

**Comment by biodranik** [comment]
Right. And @rtsisyk knew that.

Docs: https://accrescent.app/docs/guide/getting-started/building.html

1. apk set is generated
2. apk is signed with a private OM key (organicmaps-keys repository)
3. apk is uploaded to accrescent manually via GitHub login (authorization is manually approved by accrescent admins)

**Comment by rusty-snake** [comment]
Before adding more build flavors to a crescent, we could start with keeping the existing one updated. 🙊

**Comment by rtsisyk** [comment]
> Before adding more build flavors to a crescent, we could start with keeping the existing one updated. 🙊

I agree, but we are very limited in resources to add more manual steps to the release process. It would be highly beneficial to have basic automation for integrating APK uploads to Accrescent into our release process, ideally through a Gradle plugin, similar to what we already use for Google and Huawei. Does something like that exist for Accrescent?

**Comment by rusty-snake** [comment]
Not yet: https://github.com/accrescent/parcelo/issues/286
